app/controllers/api/v1/product_details_controller.rb

```diff
@@ -0,0 +1,47 @@
+class Api::V1::ProductDetailsController < ApplicationController
+    before_action :set_product_detail, only: [:show, :update, :destroy]
+    def index
+        @product_details = ProductDetail.all
+        render json: @product_details, each_serializer: ProductDetailSerializer, status: :ok
+    end
+
+    def show
+        render json: @product_detail, serialize: ProductDetailSerializer, status: :ok
+    end
+
+  def create
+    @product_detail = ProductDetail.new(product_detail_params)
+
+    if @product_detail.save
+        render json: @product_detail, status: 201
+    else
+        render_errors(@product_detail)
+    end
+  end
+
+  def update
+    if @product_detail.update(product_detail_params)
+        render json: @product_detail, status: :ok
+    else
```

Comment:
Rather than handling update and destroy exceptions in else block why haven't you written a class for handling error exceptions?